src/Hal/Bundle/PhpMetricsCollector/Collector/PhpMetricsCollector.php

```diff
@@ -1,101 +1,122 @@
 <?php
+
 namespace Hal\Bundle\PhpMetricsCollector\Collector;
 
-use Hal\Component\Token\Tokenizer;
-use Hal\Component\Token\TokenType;
-use Hal\Metrics\Complexity\Component\McCabe\McCabe;
-use Hal\Metrics\Complexity\Text\Halstead\Halstead;
-use Hal\Metrics\Complexity\Text\Length\Loc;
-use Hal\Metrics\Design\Component\MaintainabilityIndex\MaintainabilityIndex;
+use Hal\Application\Analyze;
+use Hal\Application\Config\Config;
+use Hal\Component\Issue\Issuer;
+use Hal\Component\Output\TestOutput;
+use Hal\Metric\ClassMetric;
+use Hal\Metric\Consolidated;
+use Hal\Metric\InterfaceMetric;
+use Hal\Metric\Metric;
+use Hal\Violation\Violations;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\HttpKernel\Kernel;
 
 class PhpMetricsCollector extends DataCollector
 {
-    public function collect(Request $request, Response $response, \Exception $exception = null)
+    /** @var Kernel */
+    private $kernel;
+
+    public function __construct(Kernel $kernel)
+    {
+        $this->kernel = $kernel;
+    }
+
+    public function reset()
     {
+        $this->data = [
+            'average' => [],
+            'cclasses' => 0,
+            'classes' => [],
+        ];
+    }
 
+    public function collect(Request $request, Response $response, \Throwable $exception = null)
+    {
         // filter loaded files
         $files = get_included_files();
-        $files = array_filter($files, function ($v) {
-            $excludes = array('vendor', 'pear', '\\.phar', 'bootstrap\.php', 'Test', '/app', 'AppKernel.php', 'autoload.php', 'cache/', 'app.php', 'app_dev.php', 'Form', 'PhpMetrics', 'classes.php');
-            return !preg_match('!' . implode('|', $excludes) . '!', $v);
+        $excludes = [
+            'vendor',
+            'pear',
+            '\\.phar',
```

Comment:
You should not escape this now as you may not know how this value could be use. Do not escape here, but escape when you need (when used in `preg_*`, escape with `preg_quote`)